### ISSUE TITLE
Words or phrases I don't know translations for

### ISSUE THREAD

**Issue by erfannoury** [erfannoury/cheatsheet-translation#3]
- [ ] Variance -> واریانس
- [ ] Jointly-distributed random variables
- [x] Marginal density -> چگالی حاشیه‌ای
- [ ] Covariance
- [x] Parameter -> عامل
- [x] Bias -> پیش‌قدر
- [x] Scalar -> اسکالر
- [x] Silhouette -> نیم‌رخ
- [x]  Classification -> دسته‌بندی
- [x] Regression -> وایازش
- [x] Likelihood -> درست‌نمایی
- [x] Rule (as in update rule) -> رویه
- [ ] Softmax
- [x] Statistic -> آماره
- [x] Kernel -> هسته
- [x] Confusion Matrix -> ماتریس درهم‌ریختگی
- [x] Cross-validation -> اعتبارسنجی متقاطع
- [x] Regularization -> نظام‌بخشی
- [x] Overfit -> بیش‌برازش
- [x] Ablation/Ablative analysis -> تحلیل تقطیعی
- [x] Accuracy -> صحت
- [x] Precision -> دقت
- [x] Recall  ->‌ فراخوانی
- [x] Discriminative -> متمایزکننده
- [x] Generative -> مولد
- [x] Ensemble -> گروه
- [x] Boosting -> ترقی‌دادن
- [x] Event -> رخداد
- [x] Probably Approximately Correct -> احتمالاً تقریباً درست
- [x] Shattering -> خرد شدن
- [x] Backpropagation -> انتشار معکوس
- [x] Dropout -> برون‌اندازی
- [x] Hyperparameter -> فراعامل
- [x] Fully-connected -> تمام‌متصل
- [x] Agent (in reinforcement learning) -> عامل
- [x] Policy (in reinforcement learning) -> خط‌مشی

**Comment by mhd-karimi** [comment]
I think some of these translations like `variance` and `regression`, where the English term is a common phrase even in high school books, is not needed at all.

**Comment by m-karimi** [comment]
My previous is not relevant anymore if we are going to use the English term with the translation together.

**Comment by erfannoury** [comment]
From @reza1615:
- [ ] Bias -> سوگیری

**Comment by erfannoury** [comment]
From @kazemnejad:
- [ ] Generative -> مولد

**Comment by erfannoury** [comment]
From @AlisterTA:
- [x] ReLU -> یک‌سو ساز

**Comment by reza1615** [comment]
دوستان لطفا برای ترجمه سره‌نویسی نکنید. وایازش به جای رگرسیون واقعا ناملموس است. باید رواج واژه در ادبیات تخصصی را هم در نظر گرفت. 
همچنین واریانس در کتابهای درسی هم واریانس بود

**Comment by yassersouri** [comment]
I have to disagree with the translation of "parameter" to "عامل". I think "پارامتر" is good enough.

But "Agent (in reinforcement learning)" should be translated to "عامل"

**Comment by erfannoury** [comment]
@yassersouri I kind of agree with you, but I cannot think of a better translation. Do you have anything in mind?

**Comment by erfannoury** [comment]
@reza1615 حرف شما رو قبول دارم تا حدی، ولی این نظر رو هم دارم که باید از یه جایی سعی کنیم کلمات جدیدی رو استفاده کنیم تا بالاخره استفاده‌شون فراگیر بشن. اگر این راهنماها هم توسط افراد زیادی استفاده بشن، ممکن هست که بعضی از این ترجمه‌ها به گفتمان فارسی یادگیری ماشین راه پیدا بکنن.

**Comment by reza1615** [comment]
نکته اینجاست که اگر شما برای واژه تخصصی رشته یادگیری ماشینی که تازه است واژه گزینی کنید معقول است ولی برای واژه‌ای که ۳۰-۴۰ سال است در رشته ریاضی به فارسی آمده جایگزین انتخاب کنید اشتباه و نامفهوم است مگر آنکه فرهنگستان زبان فارسی باشید و پشتوانه صدا و سیما و آموزش و پرورش داشته باشید

**Comment by erfannoury** [comment]
الان توی ترجمه یه سری از کلمات که ترجمه‌شون شناخته شده نیست توی پرانتز ترجمه‌ی معمول هم نوشته شده. اگر فکر می‌کنید کلمات دیگه‌ای هم این تغییر رو نیاز دارن بگید تا ببینیم چطوری میشه اصلاح‌شون کرد.

**Comment by momeni** [comment]
چند پیشنهاد جای‌گزین:
Variance: وردایی
Jointly-distributed random variables: متغیرهای تصادفی با توزیع مشترک
Joint probability function: تابع احتمال مشترک
Covariance: هم‌وردایی
Parameter: دانسته
Bias: اریبی
Biased: اریب
Unbiased: نااریب
Scalar: نرده‌ای
Softmax function: تابع بیشینه‌ی هموار
Verification: درستی‌سنجی
Validation: راستی‌آزمایی
Accuracy: درستی
Function Call (in programming): فراخوانی
Recall (in machine learning): به‌یاد‌آورده
Ensemble: گردایه
Hyperparameter: فرادانسته

**Comment by reza1615** [comment]
baias = سوگیری
scalar= عددی یا قابل سنجش
parameter= پارامتر
واریانس هم در فارسی جا افتاده

**Comment by erfannoury** [comment]
<div dir="rtl">
@momeni

* چقدر ترجمه‌ی پیشنهادی برای Variance خوب هست، اون کلمه رو اصلاً یادم نبود. (الان به خاطر آوردم که کلمه‌ی ناوردایی رو داریم و خیلی دیدم). همچنین هم‌وردایی هم خیلی ترجمه‌ی خوبی هست به نظرم.

* خیلی با «دانسته» برای ترجمه‌ی Parameter موافق نیستم. فکر می‌کنم با خیلی چیزها ممکنه قاطی بشه.

* در مورد ترجمه‌ی Scalar هم تاسف می‌خورم که اون ترجمه رو فراموش کردم. یادمه خیلی توی کتاب‌ها کمیت نرده‌ای خوندم. اون رو هم عوض می‌کنم حتماً.

* بقیه‌ی پیشنهادات شما رو هم بررسی می‌کنیم و با هم دوباره صحبت می‌کنیم و در نهایت تغییرات رو توی متن‌های ترجمه‌شده اعمال می‌کنیم.

بازم ممنونم از پیشنهادات شما.

**Comment by erfannoury** [comment]
Since I think that this discussion is very important and there are not enough resources which are also easily-accessible, I intend to create and maintain this repository of a Persian glossary for Machine Learning words and phrases. 
[persian-ml-glossary](https://github.com/erfannoury/persian-ml-glossary)

I will add all the suggestions submitted to this thread to the table on that repository. But I'm not sure if a readme file on a repository is the best place to have such a glossary, but I think it's easily-accessible, easy to search, and it's easy to have discussions and versioning.

Do you agree with having this glossary? Do you have any comments/suggestions? I appreciate all of your time.
@m-karimi @momeni @reza1615 @yassersouri @kazemnejad @AlisterTA

**Comment by momeni** [comment]
<div dir="rtl">
واژه‌ی «پارامتر» بنابر زمینه‌ی کاربردش، معنی‌های متفاوتی دارد:

- یک معنی در زمینه‌ی برنامه‌نویسی و پیکربندی یک سامانه است. در این زمینه به آن **نشانوند** (آنچه نشان‌دهنده‌ی بخشی از یک پیکربندی است) گفته می‌شود که به آرگومان شباهت دارد.
- در زمینه‌ی برآورد یک آماره به آن **دانسته** (به معنی آنچه از توزیع می‌دانیم در مقابل آنچه نیازمند برآورد آن هستیم) گفته می‌شود.
- گاهی هم به دنبال برآورد پارامتری (این بار با مقداری نادانسته) از یک توزیع آماری هستیم که در این حالت به آن **فراسنج** گفته می‌شود (به معنی آنچه که سنجش و دانستن آن برای سنجش دیگر آماره‌های یک توزیع لازم است).
</div>

**Comment by mhd-karimi** [comment]
<div dir="rtl">
@erfannoury 

این که یک مخزن جداگانه به عنوان واژه‌نامه یادگیری ماشین زبان فارسی داشته باشیم خیلی خوبه، میتونیم اونجا به یک توافق برسیم و بعدش کل کلمات روی همین مخزن اعمال کنیم و به عنوان مرجع در سایر مخازن استفاده کنیم.

اما پیشنهادات:
- ترجمه کلمه scalar خیلی خوبه
- اما در مورد ترجمه Variance، من همون کلمه واریانس رو بیشتر دوست دارم چون کلمه‌ای که یه دانش آموز دوم دبیرستانی هم ازش استفاده میکنه

به نظرم میتونیم ادامه بحث رو  هم در همون repository واژه نامه ادامه بدیم.
</div>

**Comment by smmoosavi** [comment]
- [ ] coefficient ->

**Comment by erfannoury** [comment]
<div dir="rtl">
@smmoosavi فکر می‌کنم ضریب ترجمه‌ی نسبتا مناسبی باشه برای این کلمه که به نوعی ترجمه‌ی مشترکی هم برای این کلمه و هم برای multiplier هست. این دو کلمه‌ی انگلیسی هم معمولاً به صورت مشابهی به کار گرفته میشن. اگر پیشنهاد دیگه‌ای دارید مطرح کنید (بهتره که توی erfannoury/persian-ml-glossary بحث بشه البته در موردش)
